### PR TITLE
Documents the new deprecations options on the rest-api-spec

### PR DESCRIPTION
**Pull request by Mpdreamz**
Relates #41439 #38613 #35262

cc @elastic/es-clients

### CODE REVIEW

**Comment by Mpdreamz** [comment]
run elasticsearch/ci1

**Comment by Mpdreamz** [comment]
run elasticsearch-ci/1

File: rest-api-spec/README.markdown [hunk 1]
```diff
@@ -46,6 +46,72 @@ The specification contains:
...
+```
+
+#### Specific paths:
+
+```json
+{
+  "api": {
+    "documentation": "",
+    "url": {
+      "path": "/_monitoring/bulk",
```
Comment (rjernst) [inline rest-api-spec/README.markdown]: GIven `path` is removed in https://github.com/elastic/elasticsearch/pull/41452, it should be removed from here too?

File: rest-api-spec/README.markdown [hunk 2]
```diff
@@ -46,6 +46,72 @@ The specification contains:
...
+```json
+{
+  "api": {
+    "documentation": "",
+    "methods": ["GET"],
+    "url": {
+      "params": {
+        "stored_fields": {
+          "type": "list",
+          "description" : ""
```
Comment (rjernst) [inline rest-api-spec/README.markdown]: This doesn't seem to be valid json. Double quote?

File: rest-api-spec/README.markdown [hunk 3]
```diff
@@ -46,6 +46,72 @@ The specification contains:
...
+    "url": {
+      "path": "/_monitoring/bulk",
+      "paths": ["/_monitoring/bulk"],
+      "deprecated_paths" : [
+        {
+          "version" : "7.0.0",
+          "path" : "/_monitoring/{type}/bulk",
+          "description" : "Specifying types in urls has been deprecated"
+        }
+      ],
```
Comment (rjernst) [inline rest-api-spec/README.markdown]: This seems to be invalid json. We should make sure these examples are all valid.

**Review by rjernst — APPROVED** [review]
Looks good, just a couple minor things.

**Pull request merged**